File: app/src/main/java/com/eugene_poroshin/money_manager/repo/database/AccountDao.java
package com.eugene_poroshin.money_manager.repo.database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface AccountDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(AccountEntity account);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void update(AccountEntity account);

    @Delete
    void delete(AccountEntity account);

    @Query("SELECT * FROM accounts")
    LiveData<List<AccountEntity>> getAllAccounts();

    @Query("SELECT account_name FROM accounts")
    LiveData<List<String>> getAccountNames();
}

File: app/src/main/java/com/eugene_poroshin/money_manager/repo/database/AccountDao.kt
package com.eugene_poroshin.money_manager.repo.database

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface AccountDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(account: AccountEntity)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    suspend fun update(account: AccountEntity)

    @Delete
    suspend fun delete(account: AccountEntity)

    @Query("SELECT * FROM accounts")
    fun allAccounts(): LiveData<List<AccountEntity>>
}